test/Common.cpp

```diff
@@ -117,9 +117,20 @@ bool CommonOptions::parse(int argc, char const* const* argv)
 
 	po::command_line_parser cmdLineParser(argc, argv);
 	cmdLineParser.options(options);
-	po::store(cmdLineParser.run(), arguments);
+	auto parsedOptions = cmdLineParser.run();
+	po::store(parsedOptions, arguments);
 	po::notify(arguments);
 
+	for (auto const& parsedOption: parsedOptions.options)
+		if (parsedOption.position_key >= 0)
+		{
+			std::stringstream errorMessage;
+			errorMessage << "Unrecognized option: ";
+			for (auto const& token: parsedOption.original_tokens)
+				errorMessage << token;
+			throw std::runtime_error(errorMessage.str());
```

Comment:
Quoting from the original issue

>  should either ignore the -- and parse the following options - or error out on --, resp. options following it.

Just to be clear, this PR decides to "error out on --" and not parse the options right?

Reply:
Yep. ``boost::program_options`` doesn't parse anything after ``--`` as regular options, but considers all of it as "positional arguments", so parsing those - at least with ``boost::program_options`` - would be a major hassle...